File: LeetCode/May challenge/Day_3.py
from collections import Counter
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        d1 = dict(Counter(ransomNote))
        d2 = dict(Counter(magazine))
        c = 0
        for i in d1:
            if i in d2 and d1[i] <= d2[i]:
                c += 1
        if c == len(d1):
            return True
        return False

File: LeetCode/May challenge/Day 3/Day_3.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        unordered_map<char,int> temp;
        
        for(auto i : ransomNote)
            temp[i]++;
        
        for(auto t: temp)
            if(count(magazine.begin(),magazine.end(),t.first)<t.second)
                return false;
        
        return true;
    }
};
